src/linux/write_zeroes.rs

```diff
@@ -32,150 +35,282 @@ impl PunchHole for File {
 
 /// A trait for writing zeroes to a stream.
 pub trait WriteZeroes {
-    /// Write zeroes to a stream.
-    ///
-    /// Write `length` bytes of zeroes to the stream, returning how many
-    /// bytes were written.
+    /// Write up to `length` bytes of zeroes to the stream, returning how many bytes were written.
     fn write_zeroes(&mut self, length: usize) -> Result<usize>;
+
+    /// Write zeroes to the stream until `length` bytes have been written.
+    ///
+    /// This method will continuously call `write_zeroes` until the requested `length` is satisfied
```

Comment:
We shouldn't leak implementation details in the documentation (i.e. that we call `write_zeroes`). We can say instead that "... will continuously write zeroes".

Same for `write_all_zeroes_at`.